autowarefoundation/autoware.universe
# refactor(lane_change): move for lane change module to interface

## Description


This pull request refactors the lane change module of the behavior path planner to use behavior trees instead of the previous implementation. The PR moves `bt_normal`'s content to `normal` and a introduces new class `LaneChangeBTModule` that inherits from the `LaneChangeInterface` and holds a normal lane change behavior tree object. It also improves the maintainability and style of some of the existing classes and removes some unnecessary header file inclusions and files.

## Related links

None.

## Tests performed

1. Build with `COMPILE_WITH_OLD_ARCHITECTURE=FALSE`
2. Build with `COMPILE_WITH_OLD_ARCHITECTURE=TRUE`
3. Test lane change scenario with planning simulator
4. Test with CI [TIER IV internal link](https://evaluation.tier4.jp/evaluation/reports/7ff81186-7d2d-50f1-9540-97a84e65e293?project_id=prd_jt) → need to re-confirm due to conflict fix

## Notes for reviewers

1. The `lane_change_module.cpp` will be removed.
2. Lane change module's class that uses behavior tree is renamed from `LaneChangeModule` to `LaneChangeBTModule`, the actual name of the bt leave will remain `LaneChangeModule`.

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
